steward/src/commands/keys/eth/import.rs

```diff
@@ -38,39 +41,63 @@ impl Runnable for ImportEthKeyCmd {
-                let seed = mnemonic.to_seed("");
-
-                let path = config.ethereum.key_derivation_path.trim();
-                let path = path
-                    .parse::<bip32::DerivationPath>()
-                    .expect("Could not parse derivation path");
-
-                let key = bip32::XPrv::derive_from_path(seed, &path).expect("Could not derive key");
-                let key = k256::SecretKey::from(key.private_key());
-                key.to_pkcs8_der()
-                    .expect("Could not PKCS8 encod private key")
-            }
-            Err(_) => {
-                let key = rpassword::read_password_from_tty(Some("> Enter your private-key:\n"))
-                    .expect("Could not read private-key");
+        if self.key.is_some() {
+            let key = match self.key.clone() {
+                Some(private_key) => private_key,
+                None => rpassword::read_password_from_tty(Some("> Enter your private-key:\n"))
+                    .expect("Could not read private-key"),
+            };
 
-                let key: ethers::types::H256 = key.parse().expect("Could not parse private-key");
+            let key = key
+                .parse::<clarity::PrivateKey>()
+                .expect("Could not parse private-key");
 
-                let key = k256::SecretKey::from_bytes(key).expect("Could not make private key");
-                key.to_pkcs8_der()
-                    .expect("Could not PKCS8 encode private key")
-            }
-        };
+            let key = SecretKey::from_bytes(key.to_bytes()).expect("Could not convert private-key");
+
+            let key = key
+                .to_pkcs8_der()
+                .expect("Could not PKCS8 encod private key");
+
+            keystore.store(&name, &key).expect("Could not store key");
+        }
 
-        keystore.store(&name, &key).expect("Could not store key");
+        if self.key.is_none() {
+            let mnemonic = match self.mnemonic.clone() {
+                Some(mnemonic) => mnemonic,
+                None => rpassword::read_password_from_tty(Some("> Enter your bip39 mnemonic:\n"))
+                    .expect("Could not read mnemonic"),
+            };
+
+            let key = match bip32::Mnemonic::new(mnemonic.trim(), Default::default()) {
+                Ok(mnemonic) => {
+                    let seed = mnemonic.to_seed("");
+
+                    let path = config.ethereum.key_derivation_path.trim();
+                    let path = path
+                        .parse::<bip32::DerivationPath>()
+                        .expect("Could not parse derivation path");
+
+                    let key =
+                        bip32::XPrv::derive_from_path(seed, &path).expect("Could not derive key");
+                    let key = k256::SecretKey::from(key.private_key());
+                    key.to_pkcs8_der()
+                        .expect("Could not PKCS8 encod private key")
+                }
+                Err(_) => {
```

Comment:
Looking at it, it seems like this section was intended to read the private key as a fallback option? Also the above version of the private key creation seems to be using `clarity` and is duplicated from gravity bridge, whereas the version here uses `ethers`. There's no need for both, the caller should be specific about whether they're entering a mnemonic or a private key.